source/nielvisiii/academicIO.py

```diff
@@ -238,6 +239,9 @@ def __write_to_cnfg_register(bank, cnfg):
             while not(cnfg == self.cnfg[bank].read() and self.ready[bank].read()):
                 pass
 
+            # wait 500 us to apply the AI registers once configuration is modified. 12*1000/40 M = 300 us < 500 us
```

Comment:
The FPGA needs time to complete the configuration after we modify the configuration. 500us is the general time it takes to complete.